### PR TITLE
The chosen app name is now being respected

### PR DESCRIPTION
**Pull request by goldbergyoni**
* A flag --app-name which sets the root app folder name (in the future also package.json "name" property)
* Refactored generation service which is now simpler
* Component tests against the generation module

### CODE REVIEW

**Review by rluvaton — APPROVED** [review]
Well done, left some general comments, feel free to merge

File: src/code-generator/generation-logic/generate-service.ts [hunk 1]
```diff
@@ -36,15 +49,12 @@ export const generateApp = async (options: generationOptions) => {
...
-  return;
-};
+}
+async function installDependencies(targetPath: string) {
+  await execa("npm", ["install"], {
+    cwd: targetPath,
+  });
+  await execa("npx", ["turbo", "run", "build", "--continue=false"], {
+    cwd: targetPath,
+  });
```
[the remark this answers is not in the document]
Comment (goldbergyoni) [reply src/code-generator/generation-logic/generate-service.ts:59]: @rluvaton Did you mean to use the word 'pipe'? we own the process and produce the information

Why to stdout? This is what the user sees, why not just stderr

File: src/code-generator/test/non-interactive-cli.test.ts [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+
+beforeEach(async () => {
+  emptyFolderForATest = await testHelpers.createUniqueFolder();
+});
+
+afterEach(async () => {
+  await fsExtra.remove(emptyFolderForATest);
+});
+
+describe("Non-interactive CLI component tests", () => {
```
[the remark this answers is not in the document]
Comment (goldbergyoni) [reply src/code-generator/test/non-interactive-cli.test.ts:16]: @rluvaton If you wanna take this, could be great, there are framework for terminal UI tests (simulates the user clicks)

My gut feeling tells me that we should focus now on the simple terminal and concentrate our efforts on better generated code

**Comment by goldbergyoni** [comment]
@rluvaton Thanks for the review, appreciate the comments

Since this is the generator, I don't bother perfecting it rather concentrate on the generated code

**Pull request merged**